GenderPayGap.WebUI/Controllers/Admin/AdminOrganisationScopeController.cs

```diff
@@ -103,18 +110,19 @@ public async Task<IActionResult> ChangeScopePost(long id, AdminChangeScopeViewMo
                 AuditedAction.AdminChangeOrganisationScope,
                 id,
                 new Dictionary<string, string> {
-                    {"PreviousScope", previousOrganisationScope.ScopeStatus.ToString()},
+                    {"PreviousScope", currentOrganisationScope.ScopeStatus.ToString()},
                     {"NewScope", newScope.ToString()},
                     {"Reason", viewModel.Reason}
                 });
 
             return RedirectToAction("ViewScopeHistory", "AdminOrganisationScope", new {id = organisation.OrganisationId});
         }
 
-        private void RetireOldScopesForCurrentSnapshotDate(Organisation organisation)
+        private void RetireOldScopesForCurrentSnapshotDate(Organisation organisation, int year)
         {
-            IOrderedEnumerable<OrganisationScope> organisationScopesForCurrentSnapshotDate = organisation.OrganisationScopes
-                .OrderByDescending(s => s.SnapshotDate);   
+            IEnumerable<OrganisationScope> organisationScopesForCurrentSnapshotDate =
+                organisation.OrganisationScopes
+                .Where(scope => scope.SnapshotDate.Year == year);

```

Comment:
This was the bug - we were setting all scope (for all years) to Retired.
Now we're only setting all scopes for the current year to Retired

Reply:
Good spot. Thanks for fixing this